modules/issue_tracker/ajax/EditIssue.php

```diff
@@ -0,0 +1,619 @@
+//TODO: change to reflect the fact that
+//TODO: if you have a visitlabel input then you don't need to insert a session
+//also make this better
+/**
+ * Validates those fields that need it
+ * Currently PSCID and visitLabel as fk to session
+ *
+ * @param array $validateValues values to be validated
+ * @param int $issueID issue ID
+ *
+ * @throws DatabaseException
+ *
+ * @return array success with values, or error with message.
+ */
+function validateInput($validateValues, $issueID)
+{
+    $db =& Database::singleton();
+
+    $old = null;
+    if ($issueID) {
+        $old = $db->pSelect(
+            "SELECT c.PSCID, s.Visit_label from candidate c " .
+            "INNER JOIN issues i ON (i.candID = c.CandID) " .
+            "INNER JOIN session s ON (i.sessionID = s.ID) WHERE i.issueID=:issueID",
+            array('issueID' => $issueID)
+        );//inner join because you only want if it has these values.
+    }
+
+    $oldPSCID = $old['PSCID'];
+    $oldVisitLabel = $old['visitLabel'];
+
+    if ((isset($validateValues['visitLabel']) && $oldPSCID)
+        || (isset($validateValues['PSCID']) && $oldVisitLabel)
+    ) {
+        $PSCID = $oldPSCID;
+        $visitLabel = $oldVisitLabel;
+        if (isset($validateValues['PSCID'])) {
+            $PSCID = $validateValues['PSCID'];
+        }
+        if (isset($validateValues['visitLabel'])) {
+            $PSCID = $validateValues['visitLabel'];
+        }
+        $isValidSession = $db->pSelectOne(
+            "SELECT s.ID FROM candidate c " .
+            "INNER JOIN session s on (c.CandID = s.CandID) " .
+            "WHERE s.CandID=:PSCID and s.Visit_label=:visitLabel",
+            array(
+                'PSCID' => $PSCID,
+                'visitLabel' => $visitLabel,
+            )
+        );
+        if (!$isValidSession) {
+            return array(
+                'isValidSubmission' => false,
+                'invalidMessage' => 'PSCID and Visit ".
+                "Label do not match a candidate session',
+            );
+        } else {
+            return array(
+                'isValidSubmission' => true,
+                'sessionID' => $isValidSession,
+            );
+        }
+    } else if (isset($validateValues['PSCID'])) {
+        $isValidCandidate = $db->pSelectOne(
+            "SELECT CandID FROM candidate WHERE PSCID=:PSCID",
+            array(
+                'PSCID' => $validateValues['PSCID'],
+            )
+        );
+        if (!$isValidCandidate) {
+            return array(
+                'isValidSubmission' => false,
+                'invalidMessage' => 'PSCID does not match a candidate',
+            );
+        } else {
+            return array(
+                'isValidSubmission' => true,
+                'candID' => $isValidCandidate,
+            );
+        }
+    } else if (isset($validateValues['visitLabel'])) {
+        return array(
+            'isValidSubmission' => false,
+            'invalidMessage' => 'A Visit Label must ".
```

Comment:
Mixing single quotes and double quotes resulted in the message that literally looks like this `A Visit Label must ". " be accompanied by a PSCID`. The line fits in 80 chars, so concatenation is not even needed IMHO.